packages/core-modules/contracts/storage/FeatureFlag.sol

```diff
@@ -0,0 +1,34 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@synthetixio/core-contracts/contracts/errors/AccessError.sol";
+import "./Permission.sol";
+
+library FeatureFlag {
+    using SetUtil for SetUtil.AddressSet;
+
+    error FeatureUnavailable();
+
+    struct Data {
+        mapping(bytes32 => Permission.Data) featureFlags;
```

Comment:
rather than having a single property mapping `bytes32 => Permission.Data`, have the `load` function take a parameter `bytes32` and then add to the `abi.encode`. Then, load that as a slot directly.

better pattern when you look at the code, saves a sidgen of gas

Reply:
in fact, you could probably just have `Permission` on its own and more or less completely remove this file (or do the opposite, move the `Permission` structure into this file and completely remove the current mapping)

Reply:
ooooh smart!  great idea, thank you!!